providers/linux/arch_linux.go

```diff
@@ -38,3 +48,33 @@ func Architecture() (string, error) {
 
 	return string(data), nil
 }
+
+func NativeArchitecture() (string, error) {
+	data, err := os.ReadFile(procSysKernelArch)
+	if err != nil {
+		if os.IsNotExist(err) {
```

Comment:
Could we use the data from `uname -m` as the fallback? That would be more general in that it handles many architecture names instead of only intel and arm. There's an example of fetching this data in the `KernelVersion()` function.

https://github.com/elastic/go-sysinfo/blob/44e5479dbc4cbc941ec76134bff5a3bfa6d6e5f4/providers/linux/kernel_linux.go#L26-L29

Reply:
that's what `func Architecture()` does, the syscall returns the emulated architecture